common/filesystem.h

```diff
@@ -844,13 +932,17 @@ class DirRef::Reader : public DirRef {
   Reader() = default;
   Reader(Reader&& arg) noexcept : DirRef(arg.dfd_), dirp_(arg.dirp_) {
     arg.dirp_ = nullptr;
+    // The directory file descriptor isn't owning, but clear it for clarity.
+    arg.dfd_ = -1;
   }
```

Comment:
```suggestion
  Reader(Reader&& arg) noexcept
      // The directory file descriptor isn't owning, but clear it for clarity.
      : DirRef(std::exchange(arg.dfd_, -1)),
      dirp_(std::exchange(arg.dirp_, nullptr)) {}
```

In line with the other exchange suggestions, capturing that this one matches the same pattern

Reply:
Done.